### PR TITLE
task4590 login logout sign up

### PR DESCRIPTION
**Pull request by laam55**
https://edu-redmine.framgia.vn/issues/4590

### CODE REVIEW

**Comment by laam55** [comment]
ready

File: app/Helpers/common.php [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<?php
+
+if (! function_exists('time_from_now')) {
+    function time_from_now($date = '2018-12-25 02:35:06')
+    {
+        $locale = Session::get('website_language');
+        $datetime = new Carbon\Carbon($date);
+        $datetime->setLocale($locale);
+        $now = $datetime->now();
+        return $datetime->diffForHumans($now);
```
Comment (BunnyPi04) [inline app/Helpers/common.php:11]: thiếu dòng trắng trc return

File: app/Http/Controllers/LoginController.php [hunk 2]
```diff
@@ -3,27 +3,86 @@
...
+
+            if ($validator->fails()) {
+                return redirect()->route('home.login')->withErrors($validator->messages());
+            } else {
+
+                if (Auth::attempt(['email' => $input['email'], 'password' => $input['password']])) {
+                    \Session::flash('success_alert', __('alert.success.login'));
+
+                    return redirect()->route('home.index');
+                }else{
```
Comment (BunnyPi04) [inline app/Http/Controllers/LoginController.php]: convention

File: app/Http/Controllers/LoginController.php [hunk 3]
```diff
@@ -3,27 +3,86 @@
...
+            } else {
+
+                $input['password'] = bcrypt($input['password']);
+                $input['fullname'] = 'your fullname';
+
+                if ($this->userRepository->create($input)) {
+                    \Session::flash('success_alert', __('alert.success.login'));
+
+                    return redirect()->route('home.index');
+                }else{
```
Comment (BunnyPi04) [inline app/Http/Controllers/LoginController.php]: convention

File: app/Repositories/Eloquents/UserRepository.php [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+namespace App\Repositories\Eloquents;
+
+use App\Repositories\Contracts\UserRepositoryInterface;
+use App\User;
+use Validator;
+
+class UserRepository implements UserRepositoryInterface
+{
+
+    protected $fillable = ['id', 'username', 'fullname', 'email', 'image_path', 'address', 'phone'];
```
Comment (BunnyPi04) [inline app/Repositories/Eloquents/UserRepository.php]: convention

File: resources/views/home/layout/header.blade.php [hunk 5]
```diff
@@ -54,7 +54,9 @@
             <li class="-ctas">
                 <a href="{{ route('home.user.detail', Auth::id()) }}" class="my-profile js-gps-track wb-hat-checked" data-gps-track="profile_summary.click()">
                     <div class="gravatar-wrapper-24" title="{{ Auth::user()['fullname'] }}">
-                        <img src="#" alt="" width="24" height="24" class="-avatar js-avatar-me" title="{{ Auth::user()['fullname'] }}"></div>
+                        <img  alt="" width="24" height="24" class="-avatar js-avatar-me"
```
Comment (BunnyPi04) [inline resources/views/home/layout/header.blade.php]: dư space

File: resources/views/home/user/temp.blade.php [hunk 6]
```diff
@@ -10,10 +10,13 @@
...
-                {{ __('user.networkProfile') }}
+                {{ __('account.logout') }}
             </a>
         </div>
     </div>
-</div>
+    <div id="main-content">
+        a
+    </div>
+</div>
```
Comment (BunnyPi04) [inline resources/views/home/user/temp.blade.php]: thiếu dòng trắng cuối file

**Comment by laam55** [comment]
ready

File: app/Repositories/Eloquents/UserRepository.php [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+{
+
+    protected $fillable = [
+        'id',
+        'username',
+        'fullname',
+        'email',
+        'image_path',
+        'address',
+        'phone'
```
Comment (BunnyPi04) [inline app/Repositories/Eloquents/UserRepository.php]: convention

**Comment by laam55** [comment]
ready

**Pull request merged**